payloads/v8/channel.ts

```diff
@@ -32,35 +32,86 @@ export interface APIPartialChannel {
 }
 
 /**
- * https://discord.com/developers/docs/resources/channel#channel-object-channel-structure
+ * This interface is used to allow easy extension for other channel types. While
+ * also allowing `APIPartialChannel` to be used without breaking.
  */
-export interface APIChannel extends APIPartialChannel {
+export interface APIChannelBase<T extends ChannelType> extends APIPartialChannel {
+	type: T;
+}
+
+// TODO: update when text in voice is released
+export type TextChannelTypes = ChannelType.DM | ChannelType.GroupDM | ChannelType.GuildNews | ChannelType.GuildText;
```

Comment:
@discordjs/the-big-4 should we have this sort of types in -types? I think these fit better in places like d.js 👀

Reply:
I don't see why not.

Reply:
Fair nuff. Then, @suneettipirneni at least rename them to singular form 🥺